packages/node/src/utils/kyve/kyve.ts

```diff
@@ -0,0 +1,294 @@
+// Copyright 2020-2024 SubQuery Pte Ltd authors & contributors
+// SPDX-License-Identifier: GPL-3.0
+
+import assert from 'assert';
+import { JsonRpcSuccessResponse } from '@cosmjs/json-rpc';
+import { Registry } from '@cosmjs/proto-signing';
+import { adaptor37 } from '@cosmjs/tendermint-rpc/build/tendermint37/adaptor';
+import {
+  BlockResponse,
+  BlockResultsResponse,
+} from '@cosmjs/tendermint-rpc/build/tendermint37/responses';
+// Currently these types are not exported
+import { StorageReceipt } from '@kyvejs/protocol';
+import { Gzip } from '@kyvejs/protocol/dist/src/reactors/compression/Gzip';
+import KyveSDK, { KyveLCDClientType } from '@kyvejs/sdk';
+import { SupportedChains } from '@kyvejs/sdk/src/constants';
+import { PoolResponse } from '@kyvejs/types/lcd/kyve/query/v1beta1/pools';
+import { getLogger } from '@subql/node-core';
+import {
+  CosmosBlock,
+  CosmosEvent,
+  CosmosTransaction,
+} from '@subql/types-cosmos';
+import axios, { AxiosRequestConfig } from 'axios';
+import { BlockContent } from '../../indexer/types';
+import { LazyBlockContent, wrapCosmosMsg } from '../cosmos';
+import { BundleDetails } from './kyveTypes';
+
+const BUNDLE_TIMEOUT = 10000; //ms
+const RADIX = 10;
+
+const parseIntRadix = (value: string) => parseInt(value, RADIX);
+
+const logger = getLogger('kyve');
+
+interface UnZippedKyveBlockReponse {
+  value: { block: any; block_results: any };
+  key: string;
+}
+
+export class KyveApi {
+  private lcdClient: KyveLCDClientType;
+  private respAdaptor = adaptor37.responses;
+  private poolId: string;
+  private currentBundleId: number;
+  private cachedBlocks: UnZippedKyveBlockReponse[];
+  private cachedBundle: any; // storage Data
+  private storageUrl: string;
+
+  constructor(
+    private chainId: string,
+    storageUrl: string,
+    kyveChainId: SupportedChains = 'kyve-1',
+  ) {
+    this.storageUrl = storageUrl;
+    this.lcdClient = new KyveSDK(kyveChainId).createLCDClient();
```

Comment:
Is there an option to provide a URL here? Doesn't seem right to provide a chain id, what if there are problems with their hard coded urls